### PR TITLE
[FLINK-25767][doc] Totally translated state.md into Chinese

### PR DESCRIPTION
**Pull request by paul8263**
## What is the purpose of the change

Totally translated state.md into Chinese

## Brief change log

- docs/content.zh/docs/dev/datastream/fault-tolerance/state.md

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 1]
```diff
@@ -25,32 +25,23 @@ specific language governing permissions and limitations
...
 
 ## Keyed DataStream
 
-If you want to use keyed state, you first need to specify a key on a
-`DataStream` that should be used to partition the state (and also the records
-in the stream themselves). You can specify a key using `keyBy(KeySelector)`
-in Java/Scala API or `key_by(KeySelector)` in Python API on a `DataStream`.
-This will yield a `KeyedStream`, which then allows operations that use keyed state.
+如果你希望使用 keyed state，首先需要为`DataStream`指定 key。这个 key 用于状态分区（也会给数据流中的记录本身分区）。
+你能够使用 `DataStream` 中 Java/Scala API 的 `keyBy(KeySelector)` 或者是 Python API 的 `key_by(KeySelector)` 来指定 key。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:36]:
```suggestion
你可以使用 `DataStream` 中 Java/Scala API 的 `keyBy(KeySelector)` 或者是 Python API 的 `key_by(KeySelector)` 来指定 key。
```

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 2]
```diff
@@ -25,32 +25,23 @@ specific language governing permissions and limitations
...
+它将返回 `KeyedStream`，从而允许使用 keyed state 操作。
 
-A key selector function takes a single record as input and returns the key for
-that record. The key can be of any type and **must** be derived from
-deterministic computations.
+Key selector 函数接收单个记录作为输入，返回这条记录的 key。该 key 可以为任何类型，它**必须**能够被推算出来。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:39]: `deterministic` 是想强调 这个selector的计算是确定的、恒定的。

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 3]
```diff
@@ -85,15 +76,12 @@ keyed = words.key_by(lambda row: row[0])
 {{< /tab >}}
 {{< /tabs >}}
 
-#### Tuple Keys and Expression Keys
+#### 元组 Keys 和表达式 Keys
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:79]: Tuple keys 说的是废弃的 `#keyBy(int... fields)` API
Expression Keys 说的是 废弃的 `keyBy(String... fields)` API，这里的翻译会让人难以理解。

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 4]
```diff
@@ -85,15 +76,12 @@ keyed = words.key_by(lambda row: row[0])
...
-keys in the Java/Scala API(still not supported in the Python API). With this you can
-specify keys using tuple field indices or expressions
-for selecting fields of objects. We don't recommend using these today but you
-can refer to the Javadoc of DataStream to learn about them. Using a KeySelector
-function is strictly superior: with Java lambdas they are easy to use and they
-have potentially less overhead at runtime.
+Flink 有两种不同定义 key 的方式：Java/Scala 的元组 key 和表达式 key （Python API 仍未支持）。 
+通过这些方式你能够通过元组字段索引，或者是选择对象字段的表达式来指定 key。
+我们不推荐这样使用，但你可以参考 `DataStream` 的 Javadoc 来学习它们。 明显更优的办法是配合 Java Lambda 使用 `KeySelector`。
+它们用起来更为简单，执行效率也可能更高。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:84]: 原文的意思是用`KeySelector`明显更好，还可以配合Java Lambda 来提高`KeySelector`的易用性，而Java Lambda带来的性能额外开销很小，并不是说这种方式的“执行效率也可能更高”

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 5]
```diff
@@ -605,41 +593,27 @@ val counts: DataStream[(String, Int)] = stream
...
 ```
 
-## Operator State
+## 操作符状态 (Operator State)
 
-*Operator State* (or *non-keyed state*) is state that is bound to one
-parallel operator instance. The [Kafka Connector]({{< ref "docs/connectors/datastream/kafka" >}}) is a good motivating example for the use of
-Operator State in Flink. Each parallel instance of the Kafka consumer maintains
-a map of topic partitions and offsets as its Operator State.
+*操作符状态*（或者*非键控状态*）是绑定到一个并行操作符实例的状态。在 Flink 中使用操作符状态，[Kafka Connector]({{< ref "docs/connectors/datastream/kafka" >}})是一个很具有启发性的例子。Kafka 消费者每个并发实例维护了 topic partitions 和偏移量的 map 作为它的操作符状态。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:598]: 这里的翻译很生硬，愿意是想用Kafka的source connector是如何实现的来阐述如何用operator state。

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 6]
```diff
@@ -605,41 +593,27 @@ val counts: DataStream[(String, Int)] = stream
...
 
-The Operator State interfaces support redistributing state among parallel
-operator instances when the parallelism is changed. There are different schemes
-for doing this redistribution.
+当并行度改变的时候，操作符状态接口支持将状态重分发给各个并行操作符实例。处理重分发过程有多种不同的方案。
 
-In a typical stateful Flink Application you don't need operators state. It is
-mostly a special type of state that is used in source/sink implementations and
-scenarios where you don't have a key by which state can be partitioned.
+在典型的有状态 Flink 作业中你无需使用操作符状态。它大都用于数据源/落地端实现，以及你不需要 state 按照 key 来分区的这类场景中，作为一种特殊类型的状态使用。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:602]: 一般不会把 sink翻译成 落地。

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 7]
```diff
@@ -605,41 +593,27 @@ val counts: DataStream[(String, Int)] = stream
...
-*Broadcast State* is a special type of *Operator State*.  It was introduced to
-support use cases where records of one stream need to be broadcasted to all
-downstream tasks, where they are used to maintain the same state among all
-subtasks. This state can then be accessed while processing records of a second
-stream. As an example where broadcast state can emerge as a natural fit, one
-can imagine a low-throughput stream containing a set of rules which we want to
-evaluate against all elements coming from another stream. Having the above type
-of use cases in mind, broadcast state differs from the rest of operator states
-in that:
+*广播状态*是一种特殊的操作符状态。它用于支持流中的元素需要广播到所有下游任务的用例。在这些用例中，广播状态用于维护所有子任务相同的状态。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:608]: `Task` 怎么翻译成了 `用例`？

**Review by Myasuka — CHANGES_REQUESTED** [review]
@paul8263 首先感谢你为Flink社区的贡献，但是整体PR的翻译非常生硬，而且还存在逻辑上的错误，有种机翻的感觉，导致我也没耐心全部看完。建议再整体润色一下，参考一下目前的其他中文文档是怎么翻译的。

**Comment by paul8263** [comment]
Hi @Myasuka ,
Thank you for valuable suggestion. I've corrected some translations and please perform an another review.

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 8]
```diff
@@ -25,32 +25,23 @@ specific language governing permissions and limitations
...
-in Java/Scala API or `key_by(KeySelector)` in Python API on a `DataStream`.
-This will yield a `KeyedStream`, which then allows operations that use keyed state.
+如果你希望使用 keyed state，首先需要为`DataStream`指定 key。这个 key 用于状态分区（也会给数据流中的记录本身分区）。
+你可以使用 `DataStream` 中 Java/Scala API 的 `keyBy(KeySelector)` 或者是 Python API 的 `key_by(KeySelector)` 来指定 key。
+它将生成 `KeyedStream`，接下来允许使用 keyed state 操作。
 
-A key selector function takes a single record as input and returns the key for
-that record. The key can be of any type and **must** be derived from
-deterministic computations.
+Key selector 函数接收单条记录作为输入，返回这条记录的 key。该 key 可以为任何类型，它**必须**源于确定计算。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:39]:
```suggestion
Key selector 函数接收单条记录作为输入，返回这条记录的 key。该 key 可以为任何类型，但是它的计算产生方式**必须**是具备确定性的。
```

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 9]
```diff
@@ -85,15 +76,12 @@ keyed = words.key_by(lambda row: row[0])
...
-keys in the Java/Scala API(still not supported in the Python API). With this you can
-specify keys using tuple field indices or expressions
-for selecting fields of objects. We don't recommend using these today but you
-can refer to the Javadoc of DataStream to learn about them. Using a KeySelector
-function is strictly superior: with Java lambdas they are easy to use and they
-have potentially less overhead at runtime.
+Flink 也有两种不同定义 key 的方式：Java/Scala API 的 Tuple key 和 Expression key （Python API 仍未支持）。 
+借此你可以通过 tuple 字段索引，或者是选取对象字段的表达式来指定 key。
+如今我们不建议这样使用，但你可以参考 `DataStream` 的 Javadoc 来了解它们。 
+使用 KeySelector 函数绝对是更好的。它配合 Java Lambda 更易于使用，运行时可能具有更小的开销。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:84]:
```suggestion
使用 KeySelector 函数显然是更好的。以几乎可以忽略的额外开销为代价，结合 Java Lambda 表达式，我们可以更方便得使用KeySelector。
```

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 10]
```diff
@@ -605,41 +593,27 @@ val counts: DataStream[(String, Int)] = stream
...
 
-The Operator State interfaces support redistributing state among parallel
-operator instances when the parallelism is changed. There are different schemes
-for doing this redistribution.
+当并行度改变的时候，算子状态接口支持将状态重新分发给各并行算子实例。处理重分发过程有多种不同的方案。
 
-In a typical stateful Flink Application you don't need operators state. It is
-mostly a special type of state that is used in source/sink implementations and
-scenarios where you don't have a key by which state can be partitioned.
+在典型的有状态 Flink 应用中你无需使用算子状态。它大都作为一种特殊类型的状态使用。用于实现 source/sink，以及你没有用来对 state 分区的 key 这类场景中。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:602]:
```suggestion
在典型的有状态 Flink 应用中你无需使用算子状态。它大都作为一种特殊类型的状态使用。用于实现 source/sink，以及无法对 state 进行分区而没有主键的这类场景中。
```

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 11]
```diff
@@ -25,32 +25,23 @@ specific language governing permissions and limitations
...
+Processing]({{< ref "docs/concepts/stateful-stream-processing" >}})。
 
 ## Keyed DataStream
 
-If you want to use keyed state, you first need to specify a key on a
-`DataStream` that should be used to partition the state (and also the records
-in the stream themselves). You can specify a key using `keyBy(KeySelector)`
-in Java/Scala API or `key_by(KeySelector)` in Python API on a `DataStream`.
-This will yield a `KeyedStream`, which then allows operations that use keyed state.
+如果你希望使用 keyed state，首先需要为`DataStream`指定 key。这个 key 用于状态分区（也会给数据流中的记录本身分区）。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:35]:
```suggestion
如果你希望使用 keyed state，首先需要为`DataStream`指定 key（主键）。这个主键 用于状态分区（也会给数据流中的记录本身分区）。
```

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 12]
```diff
@@ -605,41 +593,27 @@ val counts: DataStream[(String, Int)] = stream
...
-*Operator State* (or *non-keyed state*) is state that is bound to one
-parallel operator instance. The [Kafka Connector]({{< ref "docs/connectors/datastream/kafka" >}}) is a good motivating example for the use of
-Operator State in Flink. Each parallel instance of the Kafka consumer maintains
-a map of topic partitions and offsets as its Operator State.
+*算子状态*（或者*非 keyed 状态*）是绑定到一个并行算子实例的状态。[Kafka Connector]({{< ref "docs/connectors/datastream/kafka" >}}) 是 Flink 中使用算子状态一个很具有启发性的例子。Kafka consumer 每个并行实例维护了 topic partitions 和偏移量的 map 作为它的算子状态。
 
-The Operator State interfaces support redistributing state among parallel
-operator instances when the parallelism is changed. There are different schemes
-for doing this redistribution.
+当并行度改变的时候，算子状态接口支持将状态重新分发给各并行算子实例。处理重分发过程有多种不同的方案。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:600]:
```suggestion
当并行度改变的时候，算子状态支持将状态重新分发给各并行算子实例。处理重分发过程有多种不同的方案。
```

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 13]
```diff
@@ -25,32 +25,23 @@ specific language governing permissions and limitations
...
-A key selector function takes a single record as input and returns the key for
-that record. The key can be of any type and **must** be derived from
-deterministic computations.
+Key selector 函数接收单条记录作为输入，返回这条记录的 key。该 key 可以为任何类型，它**必须**源于确定计算。
 
-The data model of Flink is not based on key-value pairs. Therefore, you do not
-need to physically pack the data set types into keys and values. Keys are
-"virtual": they are defined as functions over the actual data to guide the
-grouping operator.
+Flink 的数据模型不基于 key-value 对，因此实际上将数据集的类型打包为 key 和 value 是没有必要的。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:41]:
```suggestion
Flink 的数据模型不基于 key-value 对，因此实际上将数据集在物理上封装成 key 和 value 是没有必要的。
```

File: docs/content.zh/docs/dev/datastream/fault-tolerance/state.md [hunk 14]
```diff
@@ -85,15 +76,12 @@ keyed = words.key_by(lambda row: row[0])
...
+#### Tuple Keys 和 Expression Keys
 
-Flink also has two alternative ways of defining keys: tuple keys and expression
-keys in the Java/Scala API(still not supported in the Python API). With this you can
-specify keys using tuple field indices or expressions
-for selecting fields of objects. We don't recommend using these today but you
-can refer to the Javadoc of DataStream to learn about them. Using a KeySelector
-function is strictly superior: with Java lambdas they are easy to use and they
-have potentially less overhead at runtime.
+Flink 也有两种不同定义 key 的方式：Java/Scala API 的 Tuple key 和 Expression key （Python API 仍未支持）。
```
Comment (Myasuka) [inline docs/content.zh/docs/dev/datastream/fault-tolerance/state.md:81]: 还是建议对 Tuple keys 和 Exprssion keys 做一个解释。

**Review by Myasuka — CHANGES_REQUESTED** [review]
Thanks for the update, please take a look at my comments.

**Review by Myasuka — APPROVED** [review]
Thanks for the contribution! LGTM

**Pull request closed without merging**